supervisor/iptables_test.go

```diff
@@ -105,3 +107,103 @@ func TestSuperviseACLs(t *testing.T) {
 	}
 
 }
+
+
+//Call Supervise and we will mock AddContainer Chain here  by changing the mock defintions of newChain
+//The function should cleanup on all errors 
+func TestAddContainerChain(t *testing.T){
+	s := doNewIPTSupervisor(t)
+	containerInfo := policy.NewPUInfo("12345")
+	containerInfo.Policy.IngressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+
+	containerInfo.Policy.EgressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+	containerInfo.Runtime.SetIPAddresses(map[string]string{"bridge": "30.30.30.30"})
+	containerInfo.Policy.PolicyIPs = []string{"30.30.30.30"}
+	m := s.ipt.(TestIptablesProvider)
+
+	m.MockNewChain(t,func(table,chain string)(error){
+		if(table == "raw" ){
+			return errors.New("Failed to create raw table")
+		}
+		return nil
+	})
+	err := s.Supervise("12345",containerInfo)
+	if(err == nil){
+		t.Errorf("ignored Error from add ContainerChain:raw")
+		t.SkipNow()
+	}
+	//Check if there are now rules anymore
+	rules,err := m.ListChains("12345")
+	if( len(rules) != 0 ){
+		t.Errorf("State was not cleared after error")
+	}
+
+	
+	m.MockNewChain(t,func(table,chain string)(error){
+		if(table == "mangle"){
+			return errors.New("Failed to create raw table")
+		}
+		return nil
+	})
+	err = s.Supervise("12345",containerInfo)
+	
+	if(err == nil){
+		t.Errorf("ignored Error from addContainerChain:mangle")
+		t.SkipNow()
+	}
+	//Check if there are now rules anymore
+	rules,err = m.ListChains("12345")
+	if( len(rules) != 0 ){
+		t.Errorf("State was not cleared after error")
+	}
+}
+
+
+func TestAddChainRules(t *testing.T){
+	s := doNewIPTSupervisor(t)
+	containerInfo := policy.NewPUInfo("12345")
+	containerInfo.Policy.IngressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+
+	containerInfo.Policy.EgressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+	containerInfo.Runtime.SetIPAddresses(map[string]string{"bridge": "30.30.30.30"})
+	containerInfo.Policy.PolicyIPs = []string{"30.30.30.30"}
+	m := s.ipt.(TestIptablesProvider)
+	
+
+	m.MockAppend(t,func(table,chain string,rulespec ...string)(error){
+		if(table == "raw" && chain == "PREROUTING"){
+			return errors.New("Append Failed")
+		}
+	})
+	err := s.Supervise("12345",containerInfo)
+	if(err == nil){
+		t.Errorf("ignored Error from Append")
+		t.SkipNow()
+	}
+	//Check if there are now rules anymore
+	rules,err := m.ListChains("12345")
```

Comment:
Will implement listChains again to call the actual listchains here. 
The step is to check if we cleaned up right after getting an error